### PR TITLE
[ACLs] Prevent data collectors from seeing each other's LOIs

### PR DESCRIPTION
**Pull request by gino-m**
This PR hard codes this behavior for surveys by default. As follow-ups, we'll want to 1) add restriction server side and 2) allow survey organizers to control whether data collectors can see each other's data.

Tested manually on Pixel 7, Android 14, API 34.

Towards #2374

@lecrabe @rfontanarosa @jcqli  FYI

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/persistence/remote/RemoteDataStore.kt [hunk 1]
```diff
@@ -44,7 +44,7 @@ interface RemoteDataStore {
   suspend fun loadTermsOfService(): TermsOfService?
 
   /** Returns all LOIs in the specified survey. Main-safe. */
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/persistence/remote/RemoteDataStore.kt:46]: Let's also update the method doc while we're at it.

File: ground/src/main/java/com/google/android/ground/repository/LocationOfInterestRepository.kt [hunk 2]
```diff
@@ -52,10 +53,15 @@ constructor(
   private val remoteDataStore: RemoteDataStore,
   private val mutationSyncWorkManager: MutationSyncWorkManager,
   private val uuidGenerator: OfflineUuidGenerator,
+  private val authenticationManager: AuthenticationManager
 ) {
   /** Mirrors locations of interest in the specified survey from the remote db into the local db. */
   suspend fun syncLocationsOfInterest(survey: Survey) {
-    val lois = remoteDataStore.loadLocationsOfInterest(survey)
+    val creatorEmail = authenticationManager.getAuthenticatedUser().email
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/repository/LocationOfInterestRepository.kt:60]: Can we add a TODO here to add some check in the future to control this behavior?

File: ground/src/main/java/com/google/android/ground/persistence/remote/RemoteDataStore.kt (hunk 1 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/persistence/remote/RemoteDataStore.kt:46]: Done!

File: ground/src/main/java/com/google/android/ground/repository/LocationOfInterestRepository.kt (hunk 2 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/repository/LocationOfInterestRepository.kt:61]: Sure. It's an FR, not a bug, but I'll add one for posterity.

**Comment by gino-m** [comment]
@shobhitagarwal1612 RFAL!

**Review by shobhitagarwal1612 — APPROVED** [review]

**Pull request merged**